### ISSUE TITLE
QOL: easy way to run lint locally that matches the behavior in CI

### ISSUE THREAD

**Issue by vkuzo** [pytorch/ao#1448]
We have the CI lint rules here: https://github.com/pytorch/ao/blob/main/.github/workflows/ruff_linter.yml , currently expressed as

```
        ruff check .
        # --isolated is used to skip the allowlist at all so this applies to all files
        # please be careful when using this large changes means everyone needs to rebase
        ruff check --isolated --select F821,F823,W191
        ruff check --select F,I
        ruff format --check || {
          echo "Ruff check failed, please try again after running 'ruff format'."
          exit 1
        }
```

It would be nice to have this logic be reusable, so I can run it locally with something like `./lint_my_code.sh` and know that CI will be green.  I used to do this with `ruff format .`, but that no longer matches the CI logic, slowing down the iteration cycle on PRs that pass `ruff format .` but fail the more extensive checks.

**Comment by jerryzh168** [comment]
yeah I already added it: https://github.com/pytorch/ao/blob/main/scripts/run_ruff_fix.sh

does this work for you?

**Comment by vkuzo** [comment]
can we make both CI and that script use the same rules, so the chance of divergence from future PRs is reduced?

**Comment by jerryzh168** [comment]
> can we make both CI and that script use the same rules, so the chance of divergence from future PRs is reduced?

yeah I also commented with the same before, although I looked again and CI is doing `check` but the script is doing `fix`, we can have a arg to say we are checking or fixing though, I'll submit a PR

**Comment by jerryzh168** [comment]
Added here: https://github.com/pytorch/ao/pull/1450, will need to change CI in a separate PR I think, since CI seems to be using code from main branch

**Comment by drisspg** [comment]
Updated here: https://github.com/pytorch/ao/pull/1453
Left comment about making sure to keep isolated checks in sync.

For more context the way to do this is to use `pre-commit run`. I would highly recommend people just install these hooks

` pre-commit install` they are very lightweight

**Comment by jerryzh168** [comment]
@drisspg are you sure this works? I just ran `pre-commit run` and nothing happens

**Comment by jerryzh168** [comment]
still some discrepancies right now I think, since ignore list: https://github.com/pytorch/ao/blob/00a8d290aab354985fce8c880e1fded22bc48e30/.github/workflows/ruff_linter.yml#L71 and https://github.com/pytorch/ao/blob/00a8d290aab354985fce8c880e1fded22bc48e30/ruff.toml#L24 are not the same, how do we align these two?

**Comment by drisspg** [comment]
you need file staged for it to run

**Comment by jerryzh168** [comment]
yeah, works now, I was a bit confused about `--isolated`, `--select` etc.
